src/mozanalysis/stats/bayesian_binary.py

```diff
@@ -0,0 +1,213 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import pandas as pd
+import numpy as np
+import scipy.stats as st
+
+import mozanalysis.stats.summarize_samples as masss
+
+
+def compare_branches(
+    df, col_label, ref_branch_label='control', num_samples=10000,
+    individual_summary_quantiles=masss.default_quantiles,
+    comparative_summary_quantiles=masss.default_quantiles
+):
+    """Jointly sample conversion rates for branches then compare them.
+
+    See `compare_branches_from_agg` for more details.
+
+    Args:
+        df: a pandas DataFrame of queried experiment data in the
+            standard format.
+        col_label (str): Label for the df column contaning the metric
+            to be analyzed.
+        ref_branch_label (str, optional): String in `df['branch']` that
+            identifies the the branch with respect to which we want to
+            calculate uplifts - usually the control branch.
+        num_samples (int, optional): The number of samples to compute.
+        individual_summary_quantiles (list, optional): Quantiles to
+            determine the confidence bands on individual branch
+            statistics. Change these when making Bonferroni corrections.
+        comparative_summary_quantiles (list, optional): Quantiles to
+            determine the confidence bands on comparative branch
+            statistics (i.e. the change relative to the reference
+            branch, probably the control). Change these when making
+            Bonferroni corrections.
+
+    Returns a dictionary:
```

Comment:
this is style advocacy and not a blocking comment _but_ i'm fully converted to defining lightweight return types with [attrs](http://www.attrs.org/en/stable/overview.html) instead of ever returning a dictionary with known-in-advance keys; here's some [additional thought-leadership from glyph](https://glyph.twistedmatrix.com/2016/08/attrs.html). that looks like it could be a useful pattern here.

Reply:
As per our discussion, I'd rather use a dictionary because it's clear at first reading that it's just a data structure with no methods or other tricks up its sleeve - no need to go to a class definition to discover this.

Yes it would be nice to "DRY" and be clearer that there is consistency between return types of the various functions, but I think this dictionaries are the user-friendlier compromise given that most people will be reading not writing mozanalysis code?